packages/nc-gui/pages/[projectType]/[projectId]/index/index/[type]/[title]/[[viewTitle]].vue

```diff
@@ -15,14 +15,17 @@ const activeTab = inject(
   computed(() => ({} as TabItem)),
 )
 
-if (!project.value.id) {
-  projectLoadedHook(async () => {
-    await getMeta(route.params.title as string, true)
+/** wait until table list loads since meta load requires table list **/
+watch(
+  () => tables.value.length,
+  (nextVal) => {
+    if (!nextVal) return
+
+    getMeta(route.params.title as string, true).finally(() => (loading.value = false))
```

Comment:
This watcher will keep watching - is this intended?
The comment says "wait until" which suggests we only wait once.
If so: You can use the `until` watcher to watch once until tables have loaded :)

Reply:
What about something like following which will only trigger once

```ts
if (tables.value.length) {
  getMeta(route.params.title as string, true).finally(() => (loading.value = false))
  loading.value = false
} else {
  /** wait until table list loads since meta load requires table list **/
  watchOnce(
    () => tables.value.length,
    (nextVal) => {
      if (!nextVal) return
      getMeta(route.params.title as string, true).finally(() => (loading.value = false))
      loading.value = false
    },
  )
}
```

Reply:
That will only watch "once" so even if tables.value.length returns undefined in the watch, it will stop the watcher fully.
Instead using `until` you could check if the length is not 0 and then trigger the watcher once and be done with it =)

https://vueuse.org/shared/until/

Reply:
length will never be `undefined`